src/main/java/edu/harvard/iq/dataverse/Dataset.java

```diff
@@ -100,8 +100,8 @@
     @OrderBy("versionNumber DESC, minorVersionNumber DESC")
     private List<DatasetVersion> versions = new ArrayList<>();
 
-    @OneToOne(mappedBy = "dataset", cascade = {CascadeType.REMOVE, CascadeType.MERGE, CascadeType.PERSIST}, orphanRemoval = true)
-    private DatasetLock datasetLock;
+    @OneToMany(mappedBy = "dataset", cascade = CascadeType.ALL, orphanRemoval = true)
+    private Set<DatasetLock> datasetLocks;
```

Comment:
Ah ha! This change to support multiple locks is what @sekmiller and I believe we need for #4139 to fix a bug related to dataset leaving the in review state when the curator uploads a tabular file: https://github.com/IQSS/dataverse/issues/4139#issuecomment-337361146